app/rdf/query-dimension-values.ts

```diff
@@ -1,29 +1,133 @@
 import { SELECT } from "@tpluscode/sparql-builder";
 import { Literal, NamedNode, Term } from "rdf-js";
-import { cube } from "./namespace";
+import { Filters } from "../configurator";
+import { cube as cubeNs } from "./namespace";
 import { sparqlClient } from "./sparql-client";
+import { Cube, CubeDimension } from "rdf-cube-view-query";
+import { dimensionIsVersioned } from "./queries";
+import * as ns from "./namespace";
 
 interface DimensionValue {
   value: Literal | NamedNode<string>;
 }
 
+/**
+ * Formats a filter value into the right format given
+ * the datatype of the dimension
+ *
+ * Seems a bit fragile, we should find a way to directly add the ^^xsd
+ * given the datatype instead of handling everycase
+ */
+const formatFilterValue = (
+  value: string | number,
+  dimension: CubeDimension
+) => {
+  if (!dimension.datatype) {
+    return `<${value}>`;
+  } else {
+    // Seems fragile
+    if (dimension.datatype.value === ns.xsd.gYear.value) {
+      return `"${value}"^^xsd:gYear`;
+    } else if (dimension.datatype.value === ns.xsd.date.value) {
+      return `"${value}"^^xsd:date`;
+    } else if (dimension.datatype.value === ns.xsd.dateTime.value) {
+      return `"${value}"^^xsd:dateTime`;
+    } else {
+      return `"${value}"`;
+    }
+  }
+};
+
+const formatFilterIntoSparqlFilter = (
+  filter: Filters[string],
+  dimension: CubeDimension,
+  versioned: boolean,
+  index: number
+) => {
+  const suffix = versioned ? "_unversioned" : "";
+  if (filter.type === "single") {
+    return `FILTER ( (?dimension${suffix}${index} = ${formatFilterValue(
+      filter.value,
+      dimension
+    )}) )`;
+  } else if (filter.type === "multi") {
+    return `FILTER ( (?dimension${suffix}${index} in (${Object.keys(
+      filter.values
+    )
+      .map((x) => formatFilterValue(x, dimension))
+      .join(",")}) ) )`;
+  } else {
+    return "";
+  }
+};
+
 /**
  * Load dimension values.
+ *
+ * Filters on other dimensions can be passed.
+ *
  */
-export async function loadDimensionValues({
-  datasetIri,
-  dimensionIri,
-}: {
-  datasetIri: Term | undefined;
-  dimensionIri: Term | undefined;
-}): Promise<Array<Literal | NamedNode>> {
-  const query = SELECT.DISTINCT`?value`.WHERE`
-    ${datasetIri} ${cube.observationSet} ?observationSet .
-    ?observationSet ${cube.observation} ?observation .
+export async function loadDimensionValues(
+  {
+    datasetIri,
+    dimension,
+    cube,
+  }: {
+    datasetIri: Term | undefined;
+    dimension: CubeDimension;
+    cube: Cube;
+  },
+  filters?: Filters
+): Promise<Array<Literal | NamedNode>> {
+  const dimensionIri = dimension.path;
+
+  let filterList = filters ? Object.entries(filters) : [];
+  filterList = filterList.slice(
+    0,
+    filterList.findIndex(([iri]) => iri == dimensionIri?.value) + 1
+  );
```

Comment:
Why are we slicing here?

Reply:
To apply above filters onto below filters

Ex: 3 columns, ordered as follow : date, monitoring program, bathing site
In this case, 

* for date, no filter apply,
* for monitoring program, date applies
* for bathing site, date and monitoring program apply

Reply:
Ah, so they are ordered by their hierarchy on the left panel? It makes sense now 👍

Reply:
No they are not yet ordered by hierarchy, the user has to do it. But when we have hierarchies, the default order will be by hierarchy.